src/System.Private.CoreLib/shared/System/Runtime/InteropServices/CollectionsMarshal.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// An unsafe class that provides a set of methods to access the underlying data representations of collections.
+    /// </summary>
+    public static class CollectionsMarshal
+    {
+        /// <summary>
+        /// Get a <see cref="Span{T}"/> view over a <see cref="List{T}"/>'s data.
+        /// Items should not be added or removed from the <see cref="List{T}"/> while the <see cref="Span{T}"/> is in use.
+        /// </summary>
```

Comment:
nit: We should get in the habit of adding docs for parameters and return values (effectively all the fields that docs require). Something to consider going forward for new API additions (of course this applies to all future PRs and not just this one since we'd like to emphasize adding docs right in time and not accrue tech debt).

For example, see the following (which has param name and return value docs):
https://github.com/dotnet/dotnet-api-docs/blob/137fdd4f501ec90359833ad280064dbde43af7d3/xml/System.Collections/IList.xml#L213-L234
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.contains?view=netframework-4.8

cc @mairaw, @carlossanlop